scheduler-driver/src/test/java/gov/nasa/jpl/aerie/scheduler/RootfindingTest.java

```diff
@@ -0,0 +1,167 @@
+        Duration.of(50, Duration.MICROSECONDS),
+        Duration.of(50, Duration.MICROSECONDS),
+        zeroSecond,
+        thirtySecond,
+        100);
+    assertEquals(5, solution.history().getHistory().size());
+    assertEquals(new EquationSolvingAlgorithms.FunctionCoordinate<>(Duration.of(19500060, Duration.MICROSECONDS), Duration.of(39000120, Duration.MICROSECONDS)), solution.functionCoordinate());
+  }
+
+
+  @Test
+  //this is reproducing issue 1139 :
+  // f(x0) throws an exception + inf val which leads in the end to a ZeroDerivative
+  public void testSimpleDiscontinuous()
+  throws EquationSolvingAlgorithms.ZeroDerivativeException, EquationSolvingAlgorithms.NoSolutionException,
+         EquationSolvingAlgorithms.ExceededMaxIterationException, EquationSolvingAlgorithms.DivergenceException,
+         EquationSolvingAlgorithms.InfiniteDerivativeException
+  {
+    final var alg = new EquationSolvingAlgorithms.SecantDurationAlgorithm<ActivityCreationTemplate.ActivityMetadata>();
+
+    //function only discontinuous at x = 1
+    final var durationFunctionDiscontinuousAtOne =
+        new EquationSolvingAlgorithms.Function<Duration, ActivityCreationTemplate.ActivityMetadata>() {
+          @Override
+          public boolean isApproximation() {
+            return false;
+          }
+
+          @Override
+          public Duration valueAt(
+              final Duration x,
+              final EquationSolvingAlgorithms.History<Duration, ActivityCreationTemplate.ActivityMetadata> historyType)
+          throws EquationSolvingAlgorithms.DiscontinuityException
+          {
+            if (x.isEqualTo(oneSecond)) {
+              throw new EquationSolvingAlgorithms.DiscontinuityException();
+            }
+            final var ret = x.times(2);
+            historyType.add(new EquationSolvingAlgorithms.FunctionCoordinate<>(x, ret), null);
+            return ret;
+          }
+        };
+
+    final var history = new ActivityCreationTemplate.HistoryWithActivity();
+    final var solution = alg.findRoot(
+        durationFunctionDiscontinuousAtOne,
+        history,
+        oneSecond,
+        twoSecond,
+        Duration.of(3, Duration.SECONDS).plus(500, Duration.MILLISECONDS),
+        Duration.of(50, Duration.MICROSECONDS),
+        Duration.of(50, Duration.MICROSECONDS),
+        zeroSecond,
+        threeSecond,
+        10);
+    assertEquals(5, solution.history().getHistory().size());
+    assertEquals(new EquationSolvingAlgorithms.FunctionCoordinate<>(Duration.of(1750000, Duration.MICROSECONDS), Duration.of(3500000, Duration.MICROSECONDS)), solution.functionCoordinate());
+  }
+
+  private Duration pow(final Duration duration, final int power){
+    if(power == 0){
+      return Duration.of(1, Duration.MICROSECONDS);
+    }
+    final var lv = duration.in(Duration.MICROSECONDS);
+    var res = lv;
+    for(int i = 1; i < power; i++){
+      res = res * lv;
+    }
+    return Duration.of(res, Duration.MICROSECONDS);
+  }
```

Comment:
Minor, can ignore: this method can be simplified into a one liner at its callsite:

```java
final var ret = Duration.of((long) Math.pow(x.in(Duration.MICROSECONDS), 2), Duration.MICROSECONDS);
```